format/spec.md

```diff
@@ -182,6 +182,21 @@ A **`list`** is a collection of values with some element type. The element field
 
 A **`map`** is a collection of key-value pairs with a key type and a value type. Both the key field and value field each have an integer id that is unique in the table schema. Map keys are required and map values can be either optional or required. Both map keys and map values may be any type, including nested types.
 
+#### Semi-structured Types
+
+A **`variant`** is a value that stores semi-structured data. The structure and data types in a variant are not necessarily consistent across rows in a table or data file. The variant type and binary encoding are defined in the [Parquet project](https://github.com/apache/parquet-format/blob/4f208158dba80ff4bff4afaa4441d7270103dff6/VariantEncoding.md). Support for Variant is added in Iceberg v3.
```

Comment:
From the linked document

> Important
> 
> This specification is still under active development, and has not been formally adopted.

assuming Parquet-level spec if subject to change, what are the conditions to release Iceberg 3 with variant support in the spec?

Secondly, the linked document talks about shredding.
How does this interact with Iceberg field IDs in the parquet metadata?
Do all the columns share field ID, or is only the first column supposed to be annotated with the field ID?
Let's make it explicit.

Reply:
From previous discussion, the community is interested in both basic Variant type support and shredding for better performance. I can see basic variant encoding is settled - we could add additional types; I think we need finalize the shredding spec so the encoding doesn't change.

Regarding shredding, are you referring to shredded subcolumns from a Variant? I'm thinking that we can clarify in shredding spec (probably after https://github.com/apache/parquet-format/pull/461/files#diff-95f43ac21fdadae78c95da23444ed7a4036a4993e9faa2ee5d8b2c29ef6d8056). The top variant column has the field ID and the subcolumns are accessed through the path like `location.lattitude`.  
 
> Secondly, the linked document talks about shredding.
> How does this interact with Iceberg field IDs in the parquet metadata?
> Do all the columns share field ID, or is only the first column supposed to be annotated with the field ID?
> Let's make it explicit.